packages/material-ui/test/integration/Menu.test.js

```diff
@@ -1,403 +1,313 @@
-                autoFocus={itemIndex === autoFocusIndex ? true : undefined}
-              >
-                Menu Item {itemIndex}
-              </TrackingMenuItem>
-            );
-          })}
-        </Menu>,
+
+    it('[variant=selectedMenu] nothing selected', () => {
+      const { getAllByRole, getByRole } = render(
+        <OpenMenu variant="selectedMenu">
+          <MenuItem />
+          <MenuItem />
+          <MenuItem />
+        </OpenMenu>,
       );
-    };
-
-    it('[variant=menu] adds coverage for rtl and Tab with no onClose', () => {
-      // This isn't adding very meaningful coverage apart from verifying it doesn't error, but
-      // it was so close to 100% that this has value in avoiding needing to drill into coverage
-      // details to see what isn't being tested.
-      mountTrackingMenu('menu', { themeDirection: 'rtl' });
-      assert.deepEqual(contentAnchorTracker, [true, false, false]);
-      assert.deepEqual(focusTracker, [false, false, false]);
-      assert.strictEqual(menuListFocusTracker, true);
-      assert.deepEqual(tabIndexTracker, [false, false, false]);
-
-      // Adds coverage for Tab with no onClose
-      TestUtils.Simulate.keyDown(document.activeElement, {
-        key: 'Tab',
-      });
-    });
 
-    it('[variant=menu] nothing selected', () => {
-      assert.deepEqual(contentAnchorTracker, [true, false, false]);
-      assert.deepEqual(focusTracker, [false, false, false]);
-      assert.strictEqual(menuListFocusTracker, true);
-      assert.deepEqual(tabIndexTracker, [false, false, false]);
+      expect(getByRole('menu')).to.be.focused;
+      getAllByRole('menuitem').forEach(item => expect(item).to.have.property('tabIndex', -1));
     });
 
+    // no case for variant=selectedMenu
     it('[variant=menu] nothing selected, autoFocus on third', () => {
-      mountTrackingMenu('menu', { autoFocusIndex: 2 });
-      assert.deepEqual(contentAnchorTracker, [true, false, false]);
-      assert.deepEqual(focusTracker, [false, false, true]);
-      assert.strictEqual(menuListFocusTracker, true);
-      assert.deepEqual(tabIndexTracker, [false, false, false]);
-    });
+      const { getAllByRole, getByRole } = render(
+        <OpenMenu variant="menu">
+          <MenuItem />
+          <MenuItem />
+          <MenuItem autoFocus />
+        </OpenMenu>,
+      );
 
-    it('[variant=selectedMenu] nothing selected', () => {
-      mountTrackingMenu('selectedMenu');
-      assert.deepEqual(contentAnchorTracker, [true, false, false]);
-      assert.deepEqual(focusTracker, [false, false, false]);
-      assert.strictEqual(menuListFocusTracker, true);
-      assert.deepEqual(tabIndexTracker, [false, false, false]);
+      expect(getByRole('menu')).to.be.focused;
+      getAllByRole('menuitem').forEach(item => expect(item).to.have.property('tabIndex', -1));
     });
 
+    // no case for variant=menu
     it('[variant=selectedMenu] nothing selected, first index invalid', () => {
```

Comment:
This test has the same outcome as `[variant=selectedMenu] nothing selected`. Should `[variant=selectedMenu] nothing selected` focus the first item and this test shouldn't?

Reply:
> This test has the same outcome as [variant=selectedMenu] nothing selected

The content anchor is different for this test.

Reply:
Could you explain what you mean by different? I can't spot a difference visually between `menu` and `selectedMenu` for pre-selected items, no selected items or `null`-items:

https://github.com/mui-org/material-ui/blob/275a966d6ed7a479904e5c544978220e3e8d9711/test/regressions/tests/Menu/MenuContentAnchors.js
![](https://argos-screenshots-us.s3.amazonaws.com/876bb0c81d139a0038b2902544cb24db)

Reply:
`selectedMenu` is the default, so to see the variant-specific differences, you need to specify `variant="menu"`

Reply:
That makes a lot more sense:

![](https://argos-screenshots-us.s3.amazonaws.com/8c3f3d31b21a84a19e5efb5d3f12fdf2)